eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -201,6 +237,41 @@ object Helpers {
     extractShutdownScript(accept.temporaryChannelId, localFeatures, remoteFeatures, accept.upfrontShutdownScript_opt).map(script_opt => (channelFeatures, script_opt))
   }
 
+  /** Called by the initiator of a dual-funded channel. */
+  def validateParamsInitiator(nodeParams: NodeParams, channelType: SupportedChannelType, localFeatures: Features[InitFeature], remoteFeatures: Features[InitFeature], open: OpenDualFundedChannel, accept: AcceptDualFundedChannel): Either[ChannelException, (ChannelFeatures, Option[ByteVector])] = {
+    validateChannelType(open.temporaryChannelId, channelType, open.channelType_opt, accept.channelType_opt, localFeatures, remoteFeatures) match {
+      case Some(t) => return Left(t)
+      case None => // we agree on channel type
+    }
+
+    if (accept.maxAcceptedHtlcs > Channel.MAX_ACCEPTED_HTLCS) return Left(InvalidMaxAcceptedHtlcs(accept.temporaryChannelId, accept.maxAcceptedHtlcs, Channel.MAX_ACCEPTED_HTLCS))
```

Comment:
Why do we need it?

Reply:
Ha! good question. @sstone any idea? As always it's mostly the inconsistency with the single funder case which bothers me.

Reply:
We can just remove it from the single funder case :)
Maybe that was used to test edge cases on testnet a very very long time ago?
I really don't see why it would be useful today.

Reply:
It seems that in bitcoin core dust is handled the same on mainnet and testnet so I don't see why we needed it ? maybe introp issues with cln or lnd a long time ago ? anyway I think it is safe to have the same checks on mainnet and testnet.

Reply:
Removed in the previous handlers as well in https://github.com/ACINQ/eclair/pull/2247/commits/2e766f4b1e511f8b341309e6951d2b6fa96ca8ef